ui/src/components/overview/Overview.js

```diff
@@ -23,6 +26,15 @@ class Overview extends Component {
       .catch(err => {
         console.error('Fetch error: ', err)
       });
+  
+    fetch(serviceInstancesUrl)
+      .then(response => response.json())
+      .then(result => {
+        this.setState({serviceInstances: result.items});
+      })
+      .catch(err => {
+        console.error('Fetch error: ', err)
+      });
```

Comment:
Should it be not better if it is encapsulated? ( E.g `fetchServiceItems(..,)` ) Something as follows. 

```javascript 
/**
   * Add small desc
   */
  componentDidMount = () => {
    /**
     * Add small desc
     */
    function getClientItems() {
      fetch(listClientsUrl)
        .then(response => response.json())
        .then(result => {
          this.setState({mobileClients: result.items});
        })
        .catch(err => {
          console.error('Fetch error: ', err)
        });
    }

    /**
     * Add small desc
     */
    function getServiceItems() {
      fetch(serviceInstancesUrl)
        .then(response => response.json())
        .then(result => {
          this.setState({serviceInstances: result.items});
        })
        .catch(err => {
          console.error('Fetch error: ', err)
        });
    }

    getClientItems();
    getServiceItems();

  }


```

Reply:
There are some things that need improvements here alright.
I want to encapsulate these backend calls into their own module. You can view these changes as WIP.

I felt this piece of work was fine as is and a follow up after to finish the integrating

Reply:
In my point of view if we try to do improvements as this one for all PRs we will have a better quality code, and easier to keep maintained. Also, if we start to push ourselves more in this direction it will be natural and it will be done automatically. 

Because after doing the merges and the features are functional is hard to come back and to do improvements/re-factories like that.

Reply:
yeah I agree the front end code is out of place a little.
The purpose of this PR is the back end changes which is what I want to land. I don't want to start adding changes in this PR that will be followed up on further PRs. I can remove the front end changes totally if you wish?

Reply:
I did not get. It was just a suggestion/recommendation. IMHO, just extract the logic and add comments as the example will be great. However, it is up to you and the repo maintainers.